docs/docs/dev/policies/mesh-trace.md

```diff
@@ -0,0 +1,340 @@
+# Mesh Trace (beta)
+
+::: warning
+This policy uses new policy matching algorithm and is in beta state,
+it should not be mixed with [TrafficTrace](traffic-trace.md).
+:::
+
+This policy enables publishing traces to a third party tracing solution.
+
+Tracing is supported over HTTP, HTTP2, and gRPC protocols.
+You must [explicitly specify the protocol](protocol-support-in-kuma.md) for each service and data plane proxy you want to enable tracing for.
+
+Kuma currently supports the following trace exposition formats:
+
+* `zipkin` traces in this format can be sent to [many different tracing backends](https://github.com/openzipkin/openzipkin.github.io/issues/65) 
+* `datadog`
+
+::: warning
+Services still need to be instrumented to preserve the trace chain across requests made across different services.
+
+You can instrument with a language library of your choice ([for zipkin](https://zipkin.io/pages/tracers_instrumentation) and [for datadog](https://docs.datadoghq.com/tracing/setup_overview/setup/java/?tab=containers)).
+For HTTP you can also manually forward the following headers:
+
+* `x-request-id`
+* `x-b3-traceid`
+* `x-b3-parentspanid`
+* `x-b3-spanid`
+* `x-b3-sampled`
+* `x-b3-flags`
+:::
+
+## Add MeshTrace resource
+
+:::::::: tabs :options="{ useUrlFragment: false }"
+::::::: tab "Zipkin"
+:::::: tabs :options="{ useUrlFragment: false }"
+::::: tab "Kubernetes"
+:::: tabs :options="{ useUrlFragment: false }"
+::: tab "Full example"
+```yaml
+apiVersion: kuma.io/v1alpha1
+kind: MeshTrace
+metadata:
+  name: default
+  namespace: kuma-system
+  labels:
+    kuma.io/mesh: default
+spec:
+  targetRef:
+    kind: Mesh
+    name: default
+  default:
+    backends:
+      - zipkin:
+          url: http://jaeger-collector.mesh-observability:9411/api/v2/spans
+          apiVersion: httpJson
+    tags:
+      - name: team
+        literal: core
+      - name: env
+        header:
+          name: x-env
+          default: prod
+      - name: version
+        header:
+          name: x-version
+    sampling:
+      overall:
+        value: 80
+      random:
+        value: 60
+      client:
+        value: 40
+```
+:::
+::: tab "Minimal example"
```

Comment:
Probably have minimal first

Reply:
WDYT about this nested style? I wanted to have a way to show the minimal example (something that people can start with and there is not too much tinkering) and the full example so it's easy to figure out what's possible. Do you think it's worth having this structure as a "template" for future policies?